### PR TITLE
feat: link sharing funnel post page

### PR DESCRIPTION
**Pull request by rebelchris**
## Changes

Moves to SSR rendering for shared post from concept here:
https://github.com/dailydotdev/apps/pull/3941

And fixes some hydratation issues from here:
https://github.com/dailydotdev/apps/pull/3880

If we detect userID in url we will render this new section:
Screenshot 2024-12-06 at 14 42 23

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

AS-817 #done 

### Preview domain
https://as-817-sharing-funnel-post.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/hooks/user/useUserShortByIdQuery.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  const queryResult = useQuery({
+    queryKey,
+    queryFn: async (): Promise<PublicProfile> => {
+      const res = await gqlClient.request(USER_SHORT_BY_ID, { id });
+
+      return res.user;
+    },
+    staleTime: StaleTime.Default,
+    enabled: !!id,
+  });
```
Comment (omBratteng) [inline packages/shared/src/hooks/user/useUserShortByIdQuery.ts:35]: Can we disable retry when the error message is `user not found`?
image

Comment (rebelchris) [reply packages/shared/src/hooks/user/useUserShortByIdQuery.ts:35]: Basically can have no retry even :)

**Comment by capJavert** [comment]
@rebelchris ssr example https://github.com/dailydotdev/apps/pull/3941

File: packages/shared/src/graphql/users.ts [hunk 2]
```diff
@@ -12,6 +12,18 @@ import { SourceMember } from './sources';
 import type { SendType } from '../hooks';
 import { DayOfWeek } from '../lib/date';
 
+export const USER_SHORT_BY_ID = `
+  query UserShortById($id: ID!) {
```
Comment (sshanzel) [inline packages/shared/src/graphql/users.ts:16]: Do we not actually have this? 🤔

**Review by sshanzel — APPROVED** [review]
Don't have anything to add 💪 🚀

**Review by capJavert — APPROVED** [review]

**Pull request merged**